core/src/main/scala/cats/instances/queue.scala

```diff
@@ -153,3 +154,31 @@ trait QueueInstances extends cats.kernel.instances.QueueInstances {
         fa.iterator.map(_.show).mkString("Queue(", ", ", ")")
     }
 }
+
+private[instances] trait QueueInstancesBinCompat0 {
+  implicit val catsStdTraverseFilterForQueue: TraverseFilter[Queue] = new TraverseFilter[Queue] {
+    val traverse: Traverse[Queue] = cats.instances.queue.catsStdInstancesForQueue
+
+    override def mapFilter[A, B](fa: Queue[A])(f: (A) => Option[B]): Queue[B] =
+      fa.collect(Function.unlift(f))
+
+    override def filter[A](fa: Queue[A])(f: (A) => Boolean): Queue[A] = fa.filter(f)
+
+    override def collect[A, B](fa: Queue[A])(f: PartialFunction[A, B]): Queue[B] = fa.collect(f)
+
+    override def flattenOption[A](fa: Queue[Option[A]]): Queue[A] = fa.flatten
+
+    def traverseFilter[G[_], A, B](fa: Queue[A])(f: (A) => G[Option[B]])(implicit G: Applicative[G]): G[Queue[B]] =
+      fa.foldRight(Eval.now(G.pure(Queue.empty[B])))(
+          (x, xse) => G.map2Eval(f(x), xse)((i, o) => i.fold(o)(_ +: o))
+        )
+        .value
```

Comment:
We can have an implementation without using `Eval`, might even perform a little better for bigger queue's; without all those extra `Eval` wrappers.

`fa.foldRight(G.pure(Queue.empty[B]))(
          (x, xse) => G.map2(f(x), xse)((i, o) => i.fold(o)(_ +: o))
        )`

Reply:
That won't short-circuit, though, right?

Reply:
@travisbrown yes agreed it won't, but for the same reason the present implementation wouldn't. 

If by short circuit you mean `f` wouldn't have to run on the remaining Queue elements (given we are folding from the right). 

This is my understanding and derived reasoning-
`map2Eval` short-circuits: given the underlying `G` structure allows it, then the `Eval` argument skips running. 

In this case however, the Eval description is computed by running `f` on each element of the `Queue` already. The first argument to `map2Eval` is not `Lazy`. In every situation `f` will run on each element.

Please correct me If I understood this wrongly.

Reply:
I see it now, we are using the wrong `foldRight` here, queue.foldRight ends up with a while loop. This should have been-

`
        traverse.foldRight[A, G[Queue[B]]](fa, Always(G.pure(Queue.empty))) { (a, lglb) =>
          G.map2Eval(f(a), lglb)((i, o) => i.fold(o)(_ +: o))
        }
        .value`

Reply:
I see the same problem with other `traverseFilter` implementations, should I open a new issue?